### PR TITLE
feat(cx-2353): add request for price estimate CTA

### PR DESCRIPTION
**Pull request by lordkiz**
The type of this PR is: **Feature**

This PR resolves [CX-2353]

### Description
<p>
    
</p>

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Added request for price estimate CTA - kizito

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-2353]: https://artsyproduct.atlassian.net/browse/CX-2353?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Added request for price estimate CTA - kizito)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5bb1e468bd2e27029111d4fb242ec6d4ffdacdd8
</p>

File: src/app/Scenes/MyCollection/Screens/Artwork/MyCollectionArtworkInsights.tsx [hunk 1]
```diff
@@ -28,6 +30,13 @@ export const MyCollectionArtworkInsights: React.FC<MyCollectionArtworkInsightsPr
     restProps.marketPriceInsights
   )
 
+  const isPOneArtist =
+    !!artwork.artists?.find((artist) => Boolean(artist?.targetSupply?.isTargetSupply)) ??
+    !!artwork.artist?.targetSupply?.isTargetSupply ??
+    false
```
Comment (MounirDhahri) [inline src/app/Scenes/MyCollection/Screens/Artwork/MyCollectionArtworkInsights.tsx:36]: **thought:** if we will be deciding `isPOneArtist` using `artwork.artist` and not `artwork.artists`, do we need to check `artwork.artists` then?

**Review by MounirDhahri — COMMENTED** [review]
This sounds good to me, thanks @lordkiz I left one small question but that's it. I don't see anything blocking on my side

File: src/app/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/RequestForPriceEstimate.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    <Box>
+      <Button
+        testID="request-price-estimate-button"
+        onPress={() => {
+          // TODO:- CX-2355: Implement the actual email sending feature
+          trackEvent(
+            tracks.trackTappedRequestPriceEstimate(
+              artwork.internalID,
+              artwork.slug,
+              marketPriceInsights?.demandRank ?? undefined
```
[the remark this answers is not in the document]
Comment (lordkiz) [reply src/app/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/RequestForPriceEstimate.tsx:37]: demandRank can be null. And Cohesion expects number or undefined.

File: src/app/Scenes/MyCollection/Screens/Artwork/MyCollectionArtworkInsights.tsx (hunk 1 above)
Comment (lordkiz) [reply src/app/Scenes/MyCollection/Screens/Artwork/MyCollectionArtworkInsights.tsx:36]: Some artworks can have many artists/collaborations I suppose

Comment (dariakoko) [reply src/app/Scenes/MyCollection/Screens/Artwork/MyCollectionArtworkInsights.tsx:36]: **question, not blocker**: What is `artwork.artist**s**` in this context?

Comment (lordkiz) [reply src/app/Scenes/MyCollection/Screens/Artwork/MyCollectionArtworkInsights.tsx:36]: @dariakoko an array of artists that created the work

Comment (lordkiz) [reply src/app/Scenes/MyCollection/Screens/Artwork/MyCollectionArtworkInsights.tsx:36]: We can assume that we want to evaluate the work if one of the artists is a P1 artist. Eg a non-p1 artist can collaborate with a p1.

Comment (dariakoko) [reply src/app/Scenes/MyCollection/Screens/Artwork/MyCollectionArtworkInsights.tsx:36]: Understood, thanks

**Review by dariakoko — APPROVED** [review]
Looks good to me 🙏

**Review by MounirDhahri — APPROVED** [review]
Great! Let's get this merged 🎉

**Pull request merged**